File: NapakalakiRuby/lib/dice.rb
# coding: utf-8

require 'singleton'

module Napakalaki
  
  class Dice
    
    include Singleton
    
    def nextNumber
      return (Random.rand(6) + 1).to_i
    end
    
  end
  
  # d=Dice.instance
  # puts "Numero: #{d.nextNumber}" 
  
end

File: Napakalaki/Dice.java

package napakalaki;

import java.util.Random;

/**
 *
 * @author alex
 */
public class Dice {
    
    private Random generator=new Random();
    
    private Dice() {
    }
    
    public static Dice getInstance() {
        return DiceHolder.INSTANCE;
    }
    
    private static class DiceHolder {

        private static final Dice INSTANCE = new Dice();
    }
    
    public int nextNumber(){
        return (generator.nextInt(6)+1);
    }
        
}
